File: demo-vk-rest-scraping.java/src/main/java/demo/file/TextFile.java
package demo.file;

import com.google.common.base.Charsets;
import com.google.common.base.Joiner;
import com.google.common.base.Splitter;
import com.google.common.collect.Lists;
import com.google.common.io.CharSink;
import com.google.common.io.CharSource;
import com.google.common.io.Files;

import java.io.File;
import java.io.IOException;
import java.nio.charset.Charset;
import java.util.List;

public final class TextFile {

    private static final Charset CHARSET = Charsets.UTF_8;

    private final String fileName;

    public TextFile(String fileName) {
        this.fileName = "src/main/resources/" + fileName;
    }

    public String load() {
        try {
            File file = new File(fileName);
            CharSource source = Files.asCharSource(file, CHARSET);
            return source.read();
        } catch (IOException e) {
            throw new RuntimeException("Exception during loading the file: " + fileName, e);
        }
    }

    public Iterable<String> loadAsIterable() {
        return Splitter.on('\n')
                .trimResults()
                .omitEmptyStrings()
                .split(load());
    }

    public List<String> loadAsList() {
        return Lists.newArrayList(loadAsIterable());
    }

    public void save(String value) {
        try {
            File file = new File(fileName);
            CharSink sink = Files.asCharSink(file, CHARSET);
            sink.write(value);
        } catch (IOException e) {
            throw new RuntimeException("Exception during saving the file: " + fileName, e);
        }
    }

    public void save(List<String> values) {
        save(Joiner.on('\n').join(values));
    }
}

File: demo-vk-rest-scraping.kotlin/src/main/kotlin/demo/file/TextFile.kt
package demo.file

import com.google.common.base.Charsets
import com.google.common.base.Joiner
import com.google.common.base.Splitter
import com.google.common.collect.Lists
import com.google.common.io.Files
import java.io.File

class TextFile(fileName: String) {

    private val CHARSET = Charsets.UTF_8
    private val fileName: String

    init {
        this.fileName = "src/main/resources/" + fileName
    }

    fun load(): String {
        val file = File(fileName)
        val source = Files.asCharSource(file, CHARSET)
        return source.read()
    }

    fun loadAsIterable(): Iterable<String> {
        return Splitter.on('\n').trimResults().omitEmptyStrings().split(load())
    }

    fun loadAsList(): List<String> {
        return Lists.newArrayList(loadAsIterable())
    }

    fun save(value: String) {
        val file = File(fileName)
        val sink = Files.asCharSink(file, CHARSET)
        sink.write(value)
    }

    fun save(values: List<String>) {
        save(Joiner.on('\n').join(values))
    }
}